### PR TITLE
ui: scrub clusterviz map locality tiers from segment analytics

### PR DESCRIPTION
**Pull request by couchand**
Locality tiers are potentially private user data that we don't want to be collecting.  Scrub them.

### CODE REVIEW

File: pkg/ui/src/redux/analytics.ts [hunk 1]
```diff
@@ -168,12 +185,25 @@ export class AnalyticsSync {
...
             if (r.match.test(location.pathname)) {
-                path = r.replace;
+
+                // Apparently TypeScript doesn't know how to dispatch functions.
+                // If there are two function overloads defined (as with
+                // String.prototype.replace), it is unable to recognize that
+                // a union of the two types can be successfully passed in as a
+                // parameter of that function.  We have to explicitly
+                // disambiguate the types for it.
+                // See https://github.com/Microsoft/TypeScript/issues/14107
```
Comment (vilterp) [inline pkg/ui/src/redux/analytics.ts]: 😞

File: pkg/ui/src/redux/analytics.ts [hunk 2]
```diff
@@ -26,8 +26,24 @@ const defaultRedactions = [
...
+        useFunction: true, // I hate TypeScript.
+        replace: function countTiers(original: string, localities: string) {
+            const tierCount = localities.match(new RegExp("/", "g")).length;
+            let redactedLocalities = "";
+            for (let i = 0; i < tierCount; i++) {
+                redactedLocalities += "/[locality]";
+            }
+            return original.replace(localities, redactedLocalities);
+        },
+    },
```
Comment (vilterp) [inline pkg/ui/src/redux/analytics.ts]: nit: weird; this file is the only one I've seen that's (mostly) indented with 4 spaces. If you feel like it, add on another commit that reindents to two.

File: pkg/ui/src/redux/analytics.spec.ts [hunk 3]
```diff
@@ -131,6 +131,54 @@ describe("analytics listener", function() {
...
+        assert.isTrue(pageSpy.calledOnce);
+        assert.deepEqual(pageSpy.args[0][0], {
+          userId: clusterID,
+          name: expected,
+          properties: {
+            path: expected,
+          },
+        });
+      });
+    });
```
Comment (vilterp) [inline pkg/ui/src/redux/analytics.spec.ts]: nice test!

**Review by vilterp — APPROVED** [review]

File: pkg/ui/src/redux/analytics.ts (hunk 2 above)
Comment (couchand) [reply pkg/ui/src/redux/analytics.ts]: didn't want to worry about it at the moment... i'm thinking we should stop dealing with these things manually.  thoughts about moving to prettier?

Comment (vilterp) [reply pkg/ui/src/redux/analytics.ts]: I’m up for giving it a try. I know @BramGruneir did earlier and it did a bunch of weird things; may take some config.

**Pull request merged**